src/libDirectoryService/DSBlockPreProcessing.cpp

```diff
@@ -335,6 +335,7 @@ bool DirectoryService::VerifyPoWOrdering(
             return item.second == toFind;
           });
 
+      std::array<unsigned char, 32> result;
       if (it == sortedPoWSolns.cend()) {
```

Comment:
The logic is correct. But i suggest to put the switch the if else condition check. Because most of the time will run to (it != sortedPoWSolns.cend()), then the code will be more efficient.